### DISCUSSION TITLE
Module can't initialize (EC25-E)

### DISCUSSION THREAD

**Discussion by natan2525** [IchthysMaranatha/asterisk-chan-quectel#48] [Q&A] [answered]
Here is error log from asterisk:

```
[Jun  8 14:00:13] ERROR[340956]: chan_quectel.c:103 alsa_card_init: snd_pcm_open failed: No such device
[Jun  8 14:00:13] ERROR[340956]: chan_quectel.c:200 soundcard_init: Problem opening ALSA capture device hw:CARD=Android,DEV=0 
[Jun  8 14:00:13] ERROR[341006]: at_queue.c:251 at_queue_run: [quectel0] Error write command 'AT' expected response 'OK' length 3, cancel
[Jun  8 14:00:13] ERROR[341006]: chan_quectel.c:617 do_monitor_phone: [quectel0] Error adding initialization commands to queue
```

aplay -l:

```
**** List of PLAYBACK Hardware Devices ****
card 0: E205U [Superlux E205U], device 0: USB Audio [USB Audio]
  Subdevices: 0/1
  Subdevice #0: subdevice #0
card 1: Generic [HD-Audio Generic], device 3: HDMI 0 [HDMI 0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: Generic [HD-Audio Generic], device 7: HDMI 1 [PL2770QS]
  Subdevices: 0/1
  Subdevice #0: subdevice #0
card 2: Generic_1 [HD-Audio Generic], device 0: ALCS1200A Analog [ALCS1200A Analog]
  Subdevices: 0/1
  Subdevice #0: subdevice #0
card 2: Generic_1 [HD-Audio Generic], device 1: ALCS1200A Digital [ALCS1200A Digital]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 4: Android [Android], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```
arecord -l:

```
**** List of CAPTURE Hardware Devices ****
card 0: E205U [Superlux E205U], device 0: USB Audio [USB Audio]
  Subdevices: 0/1
  Subdevice #0: subdevice #0
card 2: Generic_1 [HD-Audio Generic], device 0: ALCS1200A Analog [ALCS1200A Analog]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 2: Generic_1 [HD-Audio Generic], device 2: ALCS1200A Alt Analog [ALCS1200A Alt Analog]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 3: Webcam [C922 Pro Stream Webcam], device 0: USB Audio [USB Audio]
  Subdevices: 0/1
  Subdevice #0: subdevice #0
card 4: Android [Android], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```

**Comment by manojmth** [comment]
Your ports are not correctly set, also you may be using something that does not allow direct capture of audio card, what's in your quectel.conf and show result for ls ttyU*

**Reply by natan2525** [reply]
It' works when i start asterisk as root, how to allow running as asterisk user for security?

**Reply by natan2525** [reply]
When dialing in to modem the error is as follows:
`chan_quectel.c:148 alsa_card_init: Couldn't set the new hw params: Protocol error
[Jun  8 14:31:38] ERROR[342113]: chan_quectel.c:174 alsa_card_init: sw_params: Input/output error
[Jun  8 14:31:38] ERROR[342113]: chan_quectel.c:148 alsa_card_init: Couldn't set the new hw params: Protocol error
[Jun  8 14:31:38] ERROR[342113]: chan_quectel.c:174 alsa_card_init: sw_params: Input/output error
[Jun  8 14:31:43] ERROR[342729]: chan_quectel.c:670 do_monitor_phone: [quectel0] timedout while waiting 'OK' in response to 'AT'`

**Comment by manojmth** [comment]
You have to look up how to use chan_dongle with non root user. It's the same. Permissions for using ttyusb etc

**Comment by RoEdAl** [comment, marked as answer]
* Adding the `asterisk` user to the `dialout` group is required for *Asterisk* to be able to access the module.
* Adding the `asterisk` user to the `audio` group is required for *Asterisk* to be able to access the audio device (*ALSA*).

**Reply by natan2525** [reply]
Thanks i've tried udev rules to no avail this solves the permissions issue
